jdaviz/core/freezable_state.py

```diff
@@ -54,14 +57,90 @@ def _reset_x_limits(self, *event):
 class FreezableBqplotImageViewerState(BqplotImageViewerState, FreezableState):
     linked_by_wcs = False
 
+    zoom_radius = CallbackProperty(1.0, docstring="Zoom radius")
+    zoom_center_x = CallbackProperty(0.0, docstring='x-coordinate of center of zoom box')
+    zoom_center_y = CallbackProperty(0.0, docstring='y-coordinate of center of zoom box')
+
     def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
         self.wcs_only_layers = []  # For Imviz rotation use.
+        self._during_zoom_sync = False
+        self.add_callback('zoom_radius', self._set_zoom_radius_center)
+        self.add_callback('zoom_center_x', self._set_zoom_radius_center)
+        self.add_callback('zoom_center_y', self._set_zoom_radius_center)
+        for attr in ('x_min', 'x_max', 'y_min', 'y_max'):
+            self.add_callback(attr, self._set_axes_lim)
+        super().__init__(*args, **kwargs)
 
-    def reset_limits(self, *event):
-        if self.reference_data is None:  # Nothing to do
+    @contextmanager
+    def during_zoom_sync(self):
+        self._during_zoom_sync = True
+        try:
+            yield
+        except Exception:
+            self._during_zoom_sync = False
+            raise
+        self._during_zoom_sync = False
+
+    def _set_zoom_radius_center(self, *args):
+        if self._during_zoom_sync or not hasattr(self, '_viewer') or self._viewer.shape is None:
+            return
+
+        # When WCS-linked (displayed on the sky): zoom_center_x/y and zoom_radius are in sky units,
+        # x/y_min/max are in pixels of the WCS-only layer
+        if self.linked_by_wcs:
+            image, i_ref = get_reference_image_data(self._viewer.jdaviz_app, self._viewer.reference)
+            ref_wcs = image.coords
+            cr = ref_wcs.world_to_pixel_values((self.zoom_center_x, self.zoom_center_x+abs(self.zoom_radius)),  # noqa
+                                               (self.zoom_center_y, self.zoom_center_y))
+            center_x, center_xr = cr[0]
+            center_y, center_yr = cr[1]
```

Comment:
```suggestion
            center_y = cr[1][0]
```

Reply:
Might want to add a comment like this because I had to remind myself what the inputs and outputs actually mean because the astropy doc and APE 14 aren't very clear at a glance:

```python
array([x1, x2, ...]), array([y1, y2, ...]) = ref_wcs.world_to_pixel_values([ra1_deg, ra2_deg, ...], [dec1_deg, dec2_deg, ...])
```

Reply:
Underscore is fine, because technically I don't think `center_yr` assignment is correct.